core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -139,6 +139,10 @@ protected void validate(TableMetadata base, Snapshot parent) {
       if (validateNewDeleteFiles) {
         validateNoNewDeleteFiles(base, startingSnapshotId, conflictDetectionFilter, parent);
       }
+
+      if (base.formatVersion() >= 3 && addsDeleteFiles()) {
```

Comment:
Do we really need the format version here ? Why not just using `addsDeleteFiles` flag ? (just wondering).

Reply:
I've restructured this part a bit to compute dv refs as deletes are being added. No longer applies.